java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -199,6 +201,17 @@ public B setHost(String host) {
       return self();
     }
 
+    /**
+     * Universe Domain is the domain for Google Cloud Services. It follows the format of
+     * `{ServiceName}.{UniverseDomain}`. For example, speech.googleapis.com would have a Universe
+     * Domain value of `googleapis.com` and cloudasset.test.com would have a Universe Domain of
+     * `test.com`. If this value is not set, this will default to `googleapis.com`.
+     */
+    public B setUniverseDomain(String universeDomain) {
+      this.universeDomain = universeDomain;
```

Comment:
Can you add `checkNotNull(universeDomain)` if universeDomain cannot be null?

If universeDomain is ok to be null, would you add a paragraph explaining when a caller want to set this field as null?

Reply:
universeDomain _can_ be null. The resolved universeDomain will be `googleapis.com`. I'll add a sentence (there should be no reason to manually set this field as null).

Reply:
I think I misunderstood. The inputted universeDomain can never be null, but the universeDomain that is used get resolve the final universeDomain can be (as it will set the resolved universeDomain as `googleapis.com` if the user does not touch the setter). I'll update.